lib/Runtime/Library/JavascriptFunction.cpp

```diff
@@ -2282,9 +2296,16 @@ void __cdecl _alloca_probe_16()
         }
 
         // If asm.js, make sure the base address is that of the heap buffer
-        if (isAsmJs && (instrData.bufferValue != (uint64)buffer))
+        if (instrData.bufferValue != (uint64)buffer)
         {
-            return false;
+            if (isAsmJs)
```

Comment:
i think we should make it more clear that this also will include wasm, e.g. by renaming this var